app/helpers/application_helper.rb

```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
+
+  def select_type(skill_type)
+    @character.skills.order(:name).select{ |type_skill| type_skill.type_of_skill == skill_type }
```

Comment:
You can move this:

```ruby
@character.skills.order(:name)
```

Into the model with a scope:

```ruby
scope :by_name, -> {
  order(:name)
}
```

And then on the helper:

```ruby
@character.skills.by_name.select{ |type_skill| type_skill.type_of_skill == skill_type }
```

After this you have to add a test for the scope and probably memoize the  query into another method inside the module:

```ruby
def character_skills_by_name(character)
  character.skills.order(:name)
end
```

And then call this inside the `select_type` method